src/mpi/session/session_impl.c

```diff
@@ -48,12 +47,10 @@ int MPIR_Session_init_impl(MPIR_Info * info_ptr, MPIR_Errhandler * errhandler_pt
      */
     int provided;
 
-    mpi_errno = MPII_Init_thread(NULL, NULL, MPI_THREAD_MULTIPLE, &provided, &session_ptr);
+    mpi_errno = MPII_Init_thread(NULL, NULL, thread_level, &provided, &session_ptr);
```

Comment:
`thread_level` is still uninitialized here unless the user sets it in the session info

Reply:
Indeed. I assumed `MPIR_Session_get_thread_level_from_info` will set the default.

Reply:
https://github.com/pmodels/mpich/pull/7424/commits/25a1195faf3feafd30d197a0fae5ecf716123a66

Reply:
Yes, looks good now 👍.